var/spack/repos/builtin/packages/py-ansible/package.py

```diff
@@ -26,8 +27,13 @@ class PyAnsible(PythonPackage):
     version("2.7.14", sha256="92f0be1de4f9d1c0a3a35963fb853a6d7831360fd1e734cb36d601495a71770c")
     version("2.6.20", sha256="55962e79e24a67a5534bf08aa0482d5f7322ad3f112a3ebffc4a58ae02b82277")
 
-    depends_on("python@2.7:2.8,3.5:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools@66.1.0:", type="build", when="@2.16.0:")
+
     depends_on("py-jinja2", type=("build", "run"))
+    depends_on("py-jinja2@3:", type=("build", "run"), when="@2.13.0:")
     depends_on("py-pyyaml", type=("build", "run"))
+    depends_on("py-pyyaml@5.1:", type=("build", "run"), when="^python@3.8:")
```

Comment:
```suggestion
    depends_on("py-pyyaml@5.1:", type=("build", "run"))
```

Reply:
The [requirements.txt](https://github.com/ansible/ansible/blob/386edc666ec2a053b4d576fc4b2deeb46fe492b8/requirements.txt#L7) says that `py-pyyaml@5.1:` is only required when building on top of `python@3.8:`, is there a reason we wouldn't want to respect that here?

Reply:
What the requirements.txt says:
```ini
PyYAML >= 5.1  # PyYAML 5.1 is required for Python 3.8+ support
```
What you see:

> PyYAML is required for Python 3.8+ support

What I see ([5.1.1 was the first version to add 3.8 support](https://github.com/yaml/pyyaml/blob/main/CHANGES)):

> 5.1 is required for Python 3.8+ support

What pip sees (and what Spack should also see):

```ini
PyYAML >= 5.1
```

If you look at the source code, you'll see that yaml is imported unconditionally, independent of Python version.

P.S. All of this is moot because Spack only supports Python 3.8+ anyway.

Reply:
Ah! Yep okay I see, confusingly worded but you're right.